### PR TITLE
Meshless Identity Map, Regularization for Active Cell models 

### LINKED ISSUE #78: Active Cell Mapping bug.
If we have a mapping of:

```
S*(P*m + v)
```

Where S is some weighting, P is a projection up to the full domain, and v is the background this will NOT work for the regularization. Here we are multiplying by the difference: (m - mref) and the v vector is in the wrong place.

This could either be fixed by changing the regularization to map both m and mref up separately and then subtract them, but I think that is confusing.

Also, really the active cell mapping should have a background of **zero** rather than the background conductivity... which circumvents this issue all together.

At any rate, the active cell mapping is not really a linear operator...and could lead to some difficulties in usage.

### PR DESCRIPTION
**Pull request by lheagy**
- Meshlesses Identity Map (takes nP instead of a mesh)
- Tikhonov regularization if active cells are used (don't take derivs across interfaces between active cells and not)
- testing improvements: test 1D, 2D, 3D on a random tensor mesh , also test that for a constant mref, phi_m(ref) = 0

(closes #78)

### CODE REVIEW

File: SimPEG/Maps.py [hunk 1]
```diff
@@ -118,6 +118,78 @@ def __mul__(self, val):
     def __str__(self):
         return "%s(%s,%s)" % (self.__class__.__name__, self.shape[0], self.shape[1])
 
+
+class IdentityMap_Meshless(IdentityMap):
```
Comment (lheagy) [inline SimPEG/Maps.py]: or IdentityMap_NoMesh ?

**Comment by lheagy** [comment]
@grosenkj : what do you think of this?

**Comment by grosenkj** [comment]
@lheagy : I think this looks very nice, I am for it being merged in.

**Comment by lheagy** [comment]
@rowanc1 , @sgkang : good to go??

File: SimPEG/Maps.py (hunk 1 above)
Comment (rowanc1) [inline SimPEG/Maps.py]: Can we put three extra if statements instead of a new class?

```
if type(self.mesh) is int:
    self._nP = self.mesh
    self.mesh = None
```

Could change the input to `mesh_or_nP`?

**Comment by rowanc1** [comment]
Can we get rid of the _Meshless class and just make that default behaviour in the Identity?

```
eg = IdentityMap(5)
```

instead of _meshless?

**Comment by rowanc1** [comment]
@lheagy I have updated this functionality. Let me know what you think, after the tests are run we can merge?

**Comment by lheagy** [comment]
Nice, I think this looks good.

Is there any reason that we should only take the mesh or nP (ie. not allow you to give both?). I can't think of a case off-hand where you would need / want both, but maybe there is no harm in specifying both?

**Comment by rowanc1** [comment]
I was just thinking of this. It actually might need some tweaking to error
if you specify both, because we don't actually support this functionality.

**Comment by lheagy** [comment]
would you want both in the case of an active cell model?

**Comment by lheagy** [comment]
TODO: add a test to look at the gradient when there is topography for a half-space (should be zero!)

**Comment by rowanc1** [comment]
This will conflict with regularization in PR #214. @lheagy is this good to merge in?

**Comment by lheagy** [comment]
I think it is good to go. (still need to give some thought to taking both or only one of nP, mesh. there may be cases where we do want to give both --> for dealing with topo? but this can be handled in an issue)

**Comment by lheagy** [comment]
It looks like these changes got over-written somewhere down the line...

**Comment by rowanc1** [comment]
It looks as if they got to dev?

https://github.com/simpeg/simpeg/blame/dev/SimPEG/Maps.py#L15

**Comment by lheagy** [comment]
ah, I was confused about what should be in regularization.py
